Python/pylifecycle.c

```diff
@@ -2139,6 +2145,12 @@ is_valid_fd(int fd)
     if (fd < 0) {
         return 0;
     }
+#ifdef F_GETFD
```

Comment:
macOS also supports `F_GETFD` flag also.
On my macOS, all tests are passed with changes to use `F_GETFD`.

> bpo-30225: On macOS Tiger, when stdout is redirected to a pipe and the other

macOS Tiger is 10.4, we provide installers for 10.9+ nowadays.
https://www.python.org/downloads/release/python-3100/

Reply:
So I would like to suggest changing the code to support macOS also.